pandas/tests/series/methods/test_replace.py

```diff
@@ -208,6 +208,13 @@ def test_replace_with_dict_with_bool_keys(self):
         expected = pd.Series(["yes", False, "yes"])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize("dtype", ["Int8", "Int16", "Int32", "Int64"])
+    def test_replace_int_with_na(self, dtype):
+        # GH 38267
+        result = pd.Series([0, None]).astype(dtype).replace(0, pd.NA)
+        expected = pd.Series([pd.NA, pd.NA])
```

Comment:
you also need to pass dtype here else (looking at CI output) this casts to object. That's why currently this test doesn't actually pass